### PR TITLE
[FunctionApp] Updated stacks api json to add PowerShell on Linux in Functions

### PR DESCRIPTION
**Pull request by stefanushinardi**
**Description**

The PR is to add Powershell on linux offerings https://github.com/Azure/azure-cli/issues/17657
**Testing Guide**

Verified that now the command does not throw validation error when deploying with Linux os type. And also verified creation of both consumption and elastic premium plan

sample command:
```console
az functionapp create --name powershell-linux-premium-cli-create-shinardi --os-type Linux --runtime powershell --resource-group powershell-linux-cli-shinardi --functions-version 3 --storage-account powershelllinuxcli --plan ASP-pythonscaling-8a67

Application Insights "powershell-linux-premium-cli-create-shinardi" was created for this Function App. You can visit https://portal.azure.com/#resource/subscriptions/df6d3c16-321b-4f5b-a9de-c4ed04950b97/resourceGroups/powershell-linux-cli-shinardi/providers/microsoft.insights/components/powershell-linux-premium-cli-create-shinardi/overview to view your Application Insights component
```

**History Notes**

[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by stefanushinardi** [comment]
cc: @Francisco-Gamino @AnatoliB @pragnagopa @gzuber Please review this PR. Thanks!

**Review by yungezz — APPROVED** [review]

**Review by qwordy — APPROVED** [review]

File: src/azure-cli/azure/cli/command_modules/appservice/resources/LinuxFunctionsStacks.json [hunk 1]
```diff
@@ -259,6 +259,40 @@
...
+                        "appSettingsDictionary": {
+                            "FUNCTIONS_WORKER_RUNTIME": "powershell"
+                        },
+                        "siteConfigPropertiesDictionary": {
+                            "use32BitWorkerProcess": false,
+                            "linuxFxVersion": "PowerShell|7"
+                        },
+                        "isPreview": false,
+                        "isDeprecated": false,
+                        "isHidden": false
```
Comment (pragnagopa) [inline src/azure-cli/azure/cli/command_modules/appservice/resources/LinuxFunctionsStacks.json:289]: Please use: https://github.com/pragnagopa/azure-functions-supported-runtime-stacks/blob/master/LinuxFunctionsStacks.json

We want this stack to be hidden until we roll out functions host changes to support linux consumption

**Review by pragnagopa — CHANGES_REQUESTED** [review]
Please add stack as hidden. Also, need to add tests that check powershell stack

**Comment by gzuber** [comment]
> Please add stack as hidden. Also, need to add tests that check powershell stack

Hey @stefanushinardi I sent you a message about the tests, let me know if I can help out! 😄

**Comment by stefanushinardi** [comment]
@pragnagopa @gzuber Added that tests for powershell. Thanks @Hazhzeng to help fix the local setip

**Review by gzuber — APPROVED** [review]
Looks good!

**Comment by qwordy** [comment]
@stefanushinardi Are we ready to merge?

**Review by pragnagopa — APPROVED** [review]
💯 Thanks @stefanushinardi

**Comment by stefanushinardi** [comment]
@qwordy yup, thanks!

**Pull request merged**